services/age_service.js

```diff
@@ -0,0 +1,41 @@
+const ValidationError = require(`../helpers/ValidationError.js`);
+
+//gets the dob request, calculate and return age
+const calaculateAge = (dob) => {
+  //check if dob is passed
+  if (!dob) throw new ValidationError(`Please pass date of birth to continue`);
+  // try pass dob
+  let dobformated = new Date(dob);
+  //check if dob passed is string date format
+  if (dobformated == `Invalid Date`) {
+    // convert dob to number
+    const dobinnumber = Number(dob);
+    // check if number is less than 0
+    if (dobinnumber < 1)
+      throw new ValidationError(`dob cannot be lesser than 1`);
+    // try pass dob if it's the number(milliseconds) format
+    dobformated = new Date(dobinnumber);
+    // return error is dob passed is invalid
+    if (dobformated == `Invalid Date`)
+      throw new ValidationError(`Please pass a valid date to continue`);
+  }
+  const year = computeAgeHandler(dobformated);
+  if (year < 0)
+    throw new ValidationError(
+      `Your date of birth cannot be greater than the current year`
+    );
+  return year;
+}
+
+const computeAgeHandler = (birthDate) => {
+  var today = new Date()
+  var age = today.getFullYear() - birthDate.getFullYear()
+  var m = today.getMonth() - birthDate.getMonth()
+  if (m < 0 || (m === 0 && today.getDate() < birthDate.getDate())) {
+    age--
```

Comment:
Codacy has a fix for the issue: [Insert `;`](https://app.codacy.com/gh/phiiileo/validate-age/pullRequest?prid=9953536)
```suggestion
    age--;
```